packages/aws-cdk/lib/api/hotswap/appsync-mapping-templates.ts

```diff
@@ -116,8 +120,10 @@ export async function isHotswappableAppSyncChange(
             delete sdkRequestObject.runtime;
           }
 
-          const functions = await sdk.appsync().listFunctions({ apiId: sdkRequestObject.apiId });
-          const { functionId } = functions.find((fn) => fn.name === physicalName) ?? {};
+          if (!functions.hasOwnProperty(sdkRequestObject.apiId)) {
+            functions[sdkRequestObject.apiId] = await sdk.appsync().listFunctions({ apiId: sdkRequestObject.apiId });
+          }
+          const { functionId } = functions[sdkRequestObject.apiId].find((fn) => fn.name === physicalName) ?? {};
```

Comment:
Can we include a unit test that validates that:
1. `listFunctions` is called with proper params when `sdkRequestObject.apiId` is not present
2. `listFunctions` is NOT called when `sdkRequestObject.apiId` is present
?